strax/processing/statistics.py

```diff
@@ -41,90 +23,153 @@ def highest_density_region(data, fractions_desired, only_upper_part=False, _buff
-            # it might happen that we overshoot the desired fraction. Similar
-            # to the area deciles algorithm we have now to figure out at which
-            # height we actually reached the desired fraction and store the
-            # corresponding height:
             g = fraction_desired / fraction_seen
-
-            # The following gives the true height, to get here one has to
-            # solve for h:
-            # 1. fraction_seen = sum_{i=0}^j (y_i - y_j) / a_total
-            # 2. fraction_desired = sum_{i=0}^j (y_i - h) / a_total
-            # 3. g = fraction_desired/fraction_seen
-            # j == number of seen samples
-            # n == number of total samples in distribution
-            true_height = (1 - g) * np.sum(sorted_data_max_to_j) / j + g * lowest_sample_seen
+            # Compute true height (numba)
+            true_height = _compute_true_height(np.sum(sorted_data_max_to_j), j, g, actual_lowest)
             res_amp[fi] = true_height
 
-            # Find gaps and get edges of hdr intervals:
-            ind = np.sort(max_to_min[:j])
-            gaps = np.arange(1, len(ind) + 1)
+            # Only stable_sort in Python mode
+            with numba.objmode(ind="int64[:]"):
+                ind = stable_sort(max_to_min[:j])
 
-            g0 = 0
-            g_ind = -1
+            # Rest stays in numba mode
+            gaps = np.arange(1, len(ind) + 1)
             diff = ind[1:] - ind[:-1]
             gaps = gaps[:-1][diff > 1]
 
-            if len(gaps) > _buffer_size:
-                # This signal has more boundaries than the buffer can hold
-                # hence set all entries to -1 instead.
-                res[fi, 0, :] = -1
-                res[fi, 1, :] = -1
-                fi += 1
-            else:
-                for g_ind, g in enumerate(gaps):
-                    # Loop over all gaps and get outer edges:
-                    interval = ind[g0:g]
-                    res[fi, 0, g_ind] = interval[0]
-                    res[fi, 1, g_ind] = interval[-1] + 1
-                    g0 = g
-
-                # Now we have to do the last interval:
-                interval = ind[g0:]
-                res[fi, 0, g_ind + 1] = interval[0]
-                res[fi, 1, g_ind + 1] = interval[-1] + 1
-                fi += 1
-
-        if fi == (len(fractions_desired)):
-            # Found all fractions so we are done
-            return res, res_amp
-
-    # If we end up here this might be due to an offset
-    # of the distribution with respect to zero. In that case it can
-    # happen that we do not find all desired fractions.
-    # Hence we have to enforce to compute the last step from the last
-    # lowest hight we have seen to zero.
-    # Left and right edge is by definition 0 and len(data):
+            # Process intervals with numba
+            fi, res = _process_intervals_numba(ind, gaps, fi, res, 0, _buffer_size)
+
+            if fi == len(fractions_desired):
```

Comment:
Here you might have added 1 unnecessary indent.

Reply:
i removed it

Reply:
`            if fi == len(fractions_desired):` -> `        if fi == len(fractions_desired):`